simulariumio/file_converter.py

```diff
@@ -19,24 +21,207 @@
+                f"version > 1 up to current version {BINARY_SETTINGS.VERSION}, "
+                f"found {binary_version}"
+            )
+        return {
+            "n_blocks": header_values[2],
+            "block_offsets": header_values[3::3],
+            "block_types": header_values[4::3],
+            "block_lengths": header_values[5::3],
+        }
+
+    @staticmethod
+    def _binary_block_header(
+        block_index: int,
+        binary_header: Dict[str, Any],
+        binary_data: Dict[str, Any],
+    ) -> Tuple[int, int]:
+        """
+        Parse block header of data from a .simularium binary file
+        """
+        block_offset = int(
+            binary_header["block_offsets"][block_index]
+            / BINARY_SETTINGS.BYTES_PER_VALUE
+        )
+        block_type = binary_data["int_view"][block_offset]
+        if block_type != binary_header["block_types"][block_index]:
+            raise DataError(
+                "Block type is not consistent for block "
+                f"#{block_index} : {block_type} != "
+                + binary_header["block_types"][block_index]
+            )
+        block_length = binary_data["int_view"][block_offset + 1]
+        if block_length != binary_header["block_lengths"][block_index]:
+            raise DataError(
+                "Block length is not consistent for block "
+                f"#{block_index} : {block_length} != "
+                + binary_header["block_lengths"][block_index]
+            )
+        return block_type, block_length
+
+    @staticmethod
+    def _binary_block_json(
+        block_index: int,
+        binary_header: Dict[str, Any],
+        binary_data: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        """
+        Parse JSON block from a .simularium binary file
+        """
+        block_offset = binary_header["block_offsets"][block_index]
+        block_length = binary_header["block_lengths"][block_index]
+        block_header_n_bytes = (
+            BINARY_SETTINGS.BLOCK_HEADER_N_VALUES * BINARY_SETTINGS.BYTES_PER_VALUE
+        )
+        block_offset += block_header_n_bytes
+        block_length -= block_header_n_bytes
+        traj_info_bytes = binary_data["byte_view"][
+            block_offset : block_offset + block_length
+        ]
+        return json.loads(traj_info_bytes.decode("utf-8").strip("\x00"))
+
+    @staticmethod
+    def _binary_block_spatial_data(
+        block_index: int,
+        binary_header: Dict[str, Any],
+        binary_data: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        """
+        Parse spatial data binary block from a .simularium binary file
+        """
+        block_offset = (
+            int(
+                binary_header["block_offsets"][block_index]
+                / BINARY_SETTINGS.BYTES_PER_VALUE
+            )
+            + BINARY_SETTINGS.BLOCK_HEADER_N_VALUES
+        )
+        spatial_data_version = binary_data["int_view"][block_offset]
```

Comment:
storing `binary_data["int_view"]` as some other variable name like `data_as_ints` (for example) would probably make this feel a bit more concise and readable.   because it shows up so many times